frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRenameAction.java

```diff
@@ -0,0 +1,105 @@
+public abstract class BaseRenameAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IBranchNameProvider,
+      IExpectsKeyGitMacheteRepository {
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    val branchName = getNameOfBranchUnderAction(anActionEvent);
+    val branch = branchName != null
+        ? getManagedBranchByName(anActionEvent, branchName)
+        : null;
+
+    if (branch == null) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getNonHtmlString("action.GitMachete.description.disabled.undefined.machete-branch")
+          .fmt("Rename", getQuotedStringOrCurrent(branchName)));
+    } else {
+      presentation.setDescription(
+          getNonHtmlString("action.GitMachete.BaseRenameAction.description").fmt(branch.getName()));
+    }
+  }
+
+  @Override
+  @ContinuesInBackground
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    val project = getProject(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    val currentBranchName = getNameOfBranchUnderAction(anActionEvent);
+    val branchLayout = getBranchLayout(anActionEvent);
+    val branchLayoutWriter = RuntimeBinding.instantiateSoleImplementingClass(IBranchLayoutWriter.class);
+
+    if (gitRepository == null || currentBranchName == null || branchLayout == null) {
+      return;
+    }
+
+    rename(project, gitRepository, currentBranchName, branchLayout, branchLayoutWriter);
+  }
+
+  @ContinuesInBackground
+  @UIEffect
+  @IgnoreUIThreadUnsafeCalls({"git4idea.branch.GitBrancher$renameBranch(java.lang.String, java.lang.String, java.util.List)"})
+  private static void rename(Project project,
+      GitRepository gitRepository,
+      String currentBranchName, BranchLayout branchLayout, IBranchLayoutWriter branchLayoutWriter) {
+    val gitNewBranchDialog = new GitNewBranchDialog(project, Collections.singletonList(gitRepository),
+        GitBundle.message("branches.rename.branch", currentBranchName),
+        currentBranchName,
+        /* showCheckOutOption */ false,
+        /* showResetOption */ false,
+        /* showSetTrackingOption */ false,
+        /* localConflictsAllowed */ false,
+        GitBranchOperationType.RENAME);
+
+    val options = gitNewBranchDialog.showAndGetOptions();
+
+    if (options != null) {
+      GitBrancher brancher = GitBrancher.getInstance(project);
+      Runnable renameRunnable = () -> brancher.renameBranch(currentBranchName, options.getName(),
+          Collections.singletonList(gitRepository)); // runs in background
+      new RenameBackgroundable(gitRepository,
+          branchLayout,
+          branchLayoutWriter,
+          renameRunnable,
+          currentBranchName,
+          options.getName()).queue();
```

Comment:
Yeah, probably sth like

```suggestion
      brancher.renameBranch(currentBranchName, options.getName(),
          Collections.singletonList(gitRepository)); // runs in background
```
should be enough!

Reply:
Compare to `com.virtuslab.gitmachete.frontend.actions.base.BaseCheckoutAction#actionPerformed` and its call to `git4idea.branch.GitBrancher#checkout`

Reply:
Also, add `git4idea.branch.GitBrancher#renameBranch` to background-enqueuing methods in `com.virtuslab.archunit.BackgroundTaskEnqueuingTestSuite#only_continues_in_background_methods_should_enqueue_background_tasks` -> `check`